website_event_compassion/models/sale_order_line.py

```diff
@@ -32,13 +32,16 @@ def get_donation_description(self, product):
         return super().get_donation_description(product)
 
     def _prepare_invoice_line(self, **optional_values):
-        # Propagate ambassador and event to invoice line
-        res = super()._prepare_invoice_line(**optional_values)
-        analytic = self.registration_id.compassion_event_id.analytic_id
-        res.update(
-            {
-                "user_id": self.registration_id.partner_id.id,
-                "analytic_account_id": analytic.id,
-            }
-        )
+
+        res = super(SaleOrderLine, self)._prepare_invoice_line(**optional_values)
+
+        if hasattr(self, 'registration_id') and self.registration_id:
+            res = super(SaleOrderLine, self)._prepare_invoice_line(**optional_values)
+            analytic = self.registration_id.compassion_event_id.analytic_id
```

Comment:
Would be better to set all in one the dictionary values instead of just analytic for a cleaner structure
```suggestion
           user_id = self.registration_id.partner_id.id
            analytic_id = self.registration_id.compassion_event_id.analytic_id.id
            
            res.update({
                "user_id": user_id,
                "analytic_account_id": analytic_id,
            })
```
But as it is a hotfix, i wonder how important it is to make it clean...
(same case in crowdfunding changes)